keps/sig-node/4727-reasonable-image-gc/README.md

```diff
@@ -0,0 +1,442 @@
+When the feature  is turned on, the value of `--image-gc-high-threshold` must be smaller than  value of `100 - imagefs.available`. 
+When the feature is turned off, keep the previous usage.
+
+
+### Test Plan
+
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+
+##### Unit tests
+
+
+- `pkg/kubelet/apis/config/validation`: `-:-:-` - `0%`
+
+##### Integration tests
+
+
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+
+#### Alpha
+- Feature implemented behind feature gate.
+- Unit tests passed as designed in [TestPlan](#test-plan).
+
+#### Beta
+
+- Gather feedback from developers.
+
+#### GA
+- No negative feedback.
+- Update documents to reflect the changes.
+
+### Upgrade / Downgrade Strategy
+
+
+### Version Skew Strategy
+
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ImageGCBeforeStorageEviction
+  - Components depending on the feature gate: kubelet
+
+
+###### Does enabling the feature change any default behavior?
+
+Yes.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, we can just disable the feature gate.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The constraints are respected again.
+
+###### Are there any tests for feature enablement/disablement?
+
+No.
+
+### Rollout, Upgrade and Rollback Planning
+
+
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+It's an opt-in feature for end-users and will maintain current behaviors if not set, so
+it will not impact the running workloads.
+
+###### What specific metrics should inform a rollback?
+
+No.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+Monitor the metrics
+- "kubelet_image_gc_before_storage_eviction" that contains `image-gc-threshold` and `imagefs-available` labels
```

Comment:
specifically: I think this metric should say whether the image-gc-threshold is higher than `100-imagefs-available` , which would be a breaking configuration in the future

Reply:
The value of the metric is exactly what you mean, with 1 for  image-gc-threshold greater than  `100-imagefs-available` and 0 for image-gc-threshold less than or equal to `100-imagefs-available`,  the metric labels let users   see intuitively the value of the corresponding configuration